Sources/ArcGISToolkit/Components/FeatureTemplatePicker/FeatureTemplatePicker.swift

```diff
@@ -0,0 +1,334 @@
+            for (layer, table) in layersAndTables {
+                guard includeNonCreatableFeatureTemplates || table.canAddFeature else { continue }
+                var infos = [FeatureTemplateInfo]()
+                for template in table.allTemplates {
+                    let feature = table.makeFeature(template: template)
+                    let symbol = layer.renderer?.symbol(for: feature)
+                    let image = try? await symbol?.makeSwatch(scale: UIScreen.main.scale)
+                    infos.append(
+                        FeatureTemplateInfo(
+                            layer: layer,
+                            table: table,
+                            template: template,
+                            image: image
+                        )
+                    )
+                }
+                sections.append(
+                    .init(table: table, infos: infos)
+                )
+            }
+            return sections.filter { !$0.infos.isEmpty }
+        }
+        
+        /// Re-filters based on the templates or the search text.
+        private func templatesOrSearchTextDidChange() {
+            featureTemplateSections = unfilteredFeatureTemplateSections.map {
+                $0.filtered(by: searchText)
+            }
+            .filter { !$0.infos.isEmpty }
+            showContentUnavailable = unfilteredFeatureTemplateSections.isEmpty
+            showNoTemplatesFound = !showContentUnavailable && featureTemplateSections.isEmpty
+        }
+    }
+}
+
+private extension Array<Layer> {
+    /// A flattened list of the nested layers that this array of layers may contain.
+    var flattened: [Layer] {
+        flatMap { layer in
+            guard let groupLayer = layer as? GroupLayer else { return [layer] }
+            return groupLayer.layers.flattened
+        }
+    }
+}
+
+private extension GeoModel {
+    /// All the layers in the geo model.
+    var layers: [Layer] {
+        operationalLayers
+        + (basemap?.baseLayers ?? [])
+        + (basemap?.referenceLayers ?? [])
+    }
+    
+    /// A flattened list of the layers in the geo model.
+    var flattenedLayers: [Layer] {
+        layers.flattened
+    }
+    
+    /// All the feature layers in the geo model.
+    var featureLayers: [FeatureLayer] {
+        flattenedLayers
+            .compactMap { $0 as? FeatureLayer }
+    }
+    
+    /// A list containing tuples of the feature layers and the associated
+    /// ArcGIS feature tables in the geo model.
+    var arcGISFeatureLayersAndTables: [(layer: FeatureLayer, table: ArcGISFeatureTable)] {
+        featureLayers
+            .map { (layer: $0, table: $0.featureTable) }
+            .compactMap { tuple in
+                if let table = tuple.table as? ArcGISFeatureTable {
+                    return (layer: tuple.layer, table: table)
+                } else {
+                    return nil
+                }
+            }
+    }
```

Comment:
These properties can be alphabetized.

```suggestion
    /// A list containing tuples of the feature layers and the associated
    /// ArcGIS feature tables in the geo model.
    var arcGISFeatureLayersAndTables: [(layer: FeatureLayer, table: ArcGISFeatureTable)] {
        featureLayers
            .map { (layer: $0, table: $0.featureTable) }
            .compactMap { tuple in
                if let table = tuple.table as? ArcGISFeatureTable {
                    return (layer: tuple.layer, table: table)
                } else {
                    return nil
                }
            }
    }
    
    /// All the feature layers in the geo model.
    var featureLayers: [FeatureLayer] {
        flattenedLayers
            .compactMap { $0 as? FeatureLayer }
    }
    
    /// A flattened list of the layers in the geo model.
    var flattenedLayers: [Layer] {
        layers.flattened
    }
    
    /// All the layers in the geo model.
    var layers: [Layer] {
        operationalLayers
        + (basemap?.baseLayers ?? [])
        + (basemap?.referenceLayers ?? [])
    }
```